### ISSUE TITLE
Audit our crash handler

### ISSUE THREAD

**Issue by emilio** [servo/servo#12162]
It seems we're allocating on it, and probably doing other dangerous stuff. 

Possibly in most platforms this isn't a concern (I'd expect it to be synchronized by the OS), but it is considered unsafe by POSIX allocate and another bunch of things in a signal handler.

cc @jdm

**Comment by emilio** [comment]
See https://github.com/adjivas/sig/issues/2 and https://github.com/adjivas/sig/issues/3.

**Comment by emilio** [comment]
And also https://github.com/adjivas/sig/pull/4

**Comment by metajack** [comment]
Perhaps we can preallocate a small block of memory on startup that the crash handler can use? I assume it doesn't allocate all that much.

**Comment by emilio** [comment]
Yeah, the problem is not how much memory it can allocate, is that allocating there is completely implementation-defined behavior. I don't know how much of a problem that is in practice though.

**Comment by jdm** [comment]
Right; the idea is to allocate memory outside of the signal handler at startup, and make use of it when the handler is invoked.

**Comment by emilio** [comment]
Whoops, sorry, not enough coffee this morning. Yeah, that would work, though we should see how to replace things like `format!` (probably easy) and see what `Backtrace::new` does and ensure it's well defined (probably less easy).

**Comment by nox** [comment]
@ema-fox Anything more to do about this?

**Comment by nox** [comment]
Ugh, what the hell was that completely wrong mention… I meant @emilio…

**Comment by atouchet** [comment]
sig still needs to be updated to 1.0.0.

**Comment by atouchet** [comment]
@jdm updated sig in #22259. Is there anything left to do for this?

**Comment by jdm** [comment]
Yeah, we are still quite cavalier about allocation in the signal handler.
